### PR TITLE
Java 21 Support

### PR DESCRIPTION
**Pull request by eshryane**
Migrate Whois to Java 21 LTS version.

Tested `@Aspect` / `@RetryFor` annotation is still working (set port.api=80 and Jetty will fail to start).

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-image: $CI_REGISTRY/swe-database-team/gitlab-ci/whois-build:21fc0efa
+image: $CI_REGISTRY/swe-database-team/gitlab-ci/whois-build:9704969e
```
Comment (MiguelAHM) [inline .gitlab-ci.yml:1]: Temporal commit: we have to do the change once we have the final commit

File: pom.xml [hunk 2]
```diff
@@ -818,7 +818,7 @@
             <plugin>
                 <groupId>org.jacoco</groupId>
                 <artifactId>jacoco-maven-plugin</artifactId>
-                <version>0.8.8</version>
+                <version>0.8.12</version>
```
Comment (MiguelAHM) [inline pom.xml:821]: This is enough with 0.8.11
Java 21 support is in that release

Comment (eshryane) [reply pom.xml:821]: May as well update to $LATEST

**Comment by eshryane** [comment]
Integration tests passed

File: whois-endtoend/src/test/groovy/net/ripe/db/whois/spec/update/lireditable/BaseLirEditableAttributes.groovy [hunk 3]
```diff
@@ -2,8 +2,9 @@ package net.ripe.db.whois.spec.update.lireditable
 
 
 import net.ripe.db.whois.spec.BaseQueryUpdateSpec
+import org.junit.jupiter.api.Tag
 
-@org.junit.jupiter.api.Tag("IntegrationTest")
```
Comment (eshryane) [inline whois-endtoend/src/test/groovy/net/ripe/db/whois/spec/update/lireditable/BaseLirEditableAttributes.groovy:6]: No longer necessary to use fully qualified package, as the `Tag` class doesn't exist any more (there was a tag feature in Whois but we removed it)

File: whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/SchedulerConfig.java [hunk 4]
```diff
@@ -4,23 +4,22 @@
...
 
-
 import javax.sql.DataSource;
 
 @Configuration
-@EnableAspectJAutoProxy(proxyTargetClass = true)
+// @EnableAspectJAutoProxy(proxyTargetClass = true)
 @EnableScheduling
-@EnableSchedulerLock(defaultLockAtMostFor="23h")
+@EnableSchedulerLock(defaultLockAtMostFor="23h", mode = AdviceMode.ASPECTJ)
```
Comment (eshryane) [inline whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/SchedulerConfig.java:22]: Be explicit about using AspectJ mode

File: whois-api/src/test/resources/applicationContext-api-test-message-dequeue-mock.xml [hunk 5]
```diff
@@ -10,8 +10,9 @@
             location="classpath:whois.version.properties,classpath:whois.properties"
             system-properties-mode="OVERRIDE"/>
 
-    <bean id="messageDequeue" class="org.mockito.Mockito" factory-method="mock" primary="true">
-        <constructor-arg value="net.ripe.db.whois.api.mail.dequeue.MessageDequeue"/>
+    <bean id="messageDequeue" class="net.ripe.db.whois.query.support.MockFactoryBean" primary="true">
```
Comment (eshryane) [inline whois-api/src/test/resources/applicationContext-api-test-message-dequeue-mock.xml:13]: TODO: why are we mocking this class?

File: whois-api/src/test/resources/applicationContext-api-test-update-request-handler-mock.xml [hunk 6]
```diff
@@ -10,8 +10,8 @@
             location="classpath:whois.version.properties,classpath:whois.properties"
             system-properties-mode="OVERRIDE"/>
 
-    <bean id="updateRequestHandler" class="org.mockito.Mockito" factory-method="mock" primary="true">
-        <constructor-arg value="net.ripe.db.whois.update.handler.UpdateRequestHandler"/>
+    <bean id="updateRequestHandler" class="net.ripe.db.whois.query.support.MockFactoryBean" primary="true">
```
Comment (eshryane) [inline whois-api/src/test/resources/applicationContext-api-test-update-request-handler-mock.xml:13]: TODO: why are we mocking this class?

File: whois-query/src/test/resources/applicationContext-query-test-mock.xml [hunk 7]
```diff
@@ -11,17 +11,16 @@
                 classpath:whois.properties"
             system-properties-mode="OVERRIDE"/>
 
-    <bean id="queryHandler" class="org.mockito.Mockito" factory-method="mock" primary="true">
-        <constructor-arg value="net.ripe.db.whois.query.handler.QueryHandler"/>
+    <bean id="queryHandler" class="net.ripe.db.whois.query.support.MockFactoryBean" primary="true">
```
Comment (eshryane) [inline whois-query/src/test/resources/applicationContext-query-test-mock.xml:14]: TODO: why are we mocking this class?

File: whois-scheduler/src/test/resources/applicationContext-scheduler-test-mock.xml [hunk 8]
```diff
@@ -10,12 +10,12 @@
             location="classpath:whois.version.properties,classpath:whois.properties"
             system-properties-mode="OVERRIDE"/>
 
-    <bean id="ipAccessControlListDao" class="org.mockito.Mockito" factory-method="mock" primary="true">
-        <constructor-arg value="net.ripe.db.whois.query.dao.IpAccessControlListDao"/>
+    <bean id="ipAccessControlListDao" class="net.ripe.db.whois.query.support.MockFactoryBean" primary="true">
```
Comment (eshryane) [inline whois-scheduler/src/test/resources/applicationContext-scheduler-test-mock.xml:13]: TODO: why are we mocking this class?

**Review by maggarwal13 — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (eshryane) [reply .gitlab-ci.yml]: Updated to v0.0.5 - java21, amazoncorretto, node18, firefox114

**Pull request merged**